.github/workflows/ci.yml

```diff
@@ -1314,13 +1314,18 @@ jobs:
           secret_vault_address: ${{ secrets.VAULT_ADDR }}
           secret_vault_roleId: ${{ secrets.VAULT_ROLE_ID }}
 
+  # Dynamically generate the Docker tag (e.g., SNAPSHOT or X.Y-SNAPSHOT) based on branch name
+  get-snapshot-docker-version-tag:
```

Comment:
Should this job use a `needs: [check-results]`, as otherwise it will start running directly at the start of the build when all the other CI checks start - if those fail and the effort to calculate the version tag was wasted.